skbeam/core/accumulators/arrayProjector.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+
+import math
+import numpy as np
+
+"""
+authors: Mikhail Dubrovin, TJ Lane, Christopher O'Grady
+"""
+
+class ArrayProjector(object):
+
+    def __init__(self, bin_values, weights=None, nbins=101, norm=True):
+        """
+        Parameters
+        ----------
+        bin_values : np.ndarray
+            For each pixel, this is the value of that determines which bin
+            a pixel's intensity will contribute to
+        weights : np.ndarray
+            A weight for each pixel with the same shape as bin_values (can be zero to ignore a pixel)
+        nbins : int
+            The number of bins to employ. If `None` guesses a good value.
+        """
+
+        self.bin_values = bin_values
+        self.weights = weights
+        self.nbins = nbins
+        self.norm = norm
+
+        self.binvalRange = self.bin_values.max() - self.bin_values.min()
+        self.bin_width = self.binvalRange / (float(nbins) - 1)
+
+        self._bin_assignments = np.floor( (self.bin_values - self.bin_values.min()) / self.bin_width ).astype(np.int32)
+        if weights is None:
+            self._normalization_array = (np.bincount( self._bin_assignments.flatten() ) \
+                                             + 1e-100).astype(np.float)
+        else:
+            self._normalization_array = (np.bincount( self._bin_assignments.flatten(), weights=self.weights.flatten() ) \
+                                             + 1e-100).astype(np.float)
```

Comment:
IIRC (as the original author of this code lol) the small number avoids div by zero (ie, enforces the convention 0/0 = 0)